src/prop.ts

```diff
@@ -7,6 +7,13 @@
  * @data_last
  * @category Object
  */
-export function prop<K extends string = string>(name: K) {
-  return <T extends Record<any, any>>(obj: T) => obj[name];
+
+interface PropInterface {
```

Comment:
I'd call this PropFunc, interface is just the "tool" here, it could (should?) be a type too